File: client/android/BigFoxAndroid/app/src/main/java/vn/com/onesoft/bigfox/io/core/encrypt/EncryptManager.java
package vn.com.onesoft.bigfox.io.core.encrypt;

/**
 * Created by phamquan on 8/18/15.
 */

import vn.com.onesoft.bigfox.io.core.session.ConnectionManager;

/**
 * @author QuanPH
 */
public class EncryptManager {

    public static byte[] crypt(byte[] data) {
        int validationCode = ConnectionManager.getInstance().validationCode;
        for (int i = 24; i < data.length; i++) {
            data[i] = (byte) ((data[i] ^ validationCode) & 0x00ff);
        }
        return data;
    }

}

File: client/ios/BigFoxClient/BigFoxClient/core/encrypt/EncryptManager.m
//
//  EncryptManager.m
//  BigFoxClient
//
//  Created by QuyenNX on 9/4/15.
//  Copyright (c) 2015 QuyenNX. All rights reserved.
//

#import "EncryptManager.h"
#import "ConnectionManager.h"

@implementation EncryptManager
+ (NSData*) crypt:(NSData *)data {
    if ([data length] <= 24) {
        return data;
    }else{
        char* cData = [data bytes];
        int length = [data length];
        int validationCode = [[ConnectionManager getInstance]getValidationCode];
        for (int i = 24; i < length; i++) {
            cData[i] = (char) ((cData[i]^ validationCode) & 0x00ff);
        }
        return  [NSData dataWithBytes:cData length:length];
    }
    
}
@end
